stempy/electron.cpp

```diff
@@ -114,51 +114,61 @@ std::vector<uint32_t> maximalPointsParallel(
 namespace stempy {
 
 // Implementation of modulus that "wraps" for negative numbers
+// for example, if x = 10, y = 50, x % 50 = 10
+// customized mod(10, 50) = ((10%50)+50) % 50 = 60%50 = 10
 inline uint16_t mod(uint16_t x, uint16_t y)
 {
   return ((x % y) + y) % y;
 }
 
-// Return the points in the frame with values larger than all 8 of their nearest
-// neighbors
+// Return the points in the frame with values larger than all 8 of their nearest neighbors
 std::vector<uint32_t> maximalPoints(
   const std::vector<uint16_t>& frame, int width, int height)
 {
   std::vector<uint32_t> events;
   auto numberOfPixels = height * width;
   for (int i = 0; i < numberOfPixels; i++) {
+    // int number row
     auto row = i / width;
     auto column = i % width;
+
+    // column number of left and right neighbor
+    // mod((column + 1), width) more consistent?
     auto rightNeighbourColumn = mod((i + 1), width);
     auto leftNeighbourColumn = mod((i - 1), width);
+
+    // row number of top and bottom neighbor
     auto topNeighbourRow = mod((row - 1), height);
     auto bottomNeighbourRow = mod((row + 1), height);
+
+    // pixel value at the current pixel
     auto pixelValue = frame[i];
+
+    // start index of the first element in a row
     auto bottomNeighbourRowIndex = bottomNeighbourRow * width;
     auto topNeighbourRowIndex = topNeighbourRow * width;
+
+    // is rowIndex the same as i?
     auto rowIndex = row * width;
 
-    // top
+    // if larger than top neighbor pixel value
     auto event = pixelValue > frame[topNeighbourRowIndex + column];
-    // top right
-    event =
-      event && pixelValue > frame[topNeighbourRowIndex + rightNeighbourColumn];
-    // right
+    // if larger than top right neighbor pixel value
+    event = event && pixelValue > frame[topNeighbourRowIndex + rightNeighbourColumn];
+    // if larger than right neighbor pixel value
     event = event && pixelValue > frame[rowIndex + rightNeighbourColumn];
-    // bottom right
-    event = event &&
-            pixelValue > frame[bottomNeighbourRowIndex + rightNeighbourColumn];
-    // bottom
+    // if larger than bottom right neighbor pixel value
+    event = event && pixelValue > frame[bottomNeighbourRowIndex + rightNeighbourColumn];
+    // if larger than bottom neighbor pixel value
     event = event && pixelValue > frame[bottomNeighbourRowIndex + column];
-    // bottom left
-    event = event &&
-            pixelValue > frame[bottomNeighbourRowIndex + leftNeighbourColumn];
+    // if larger than bottom left neighbor pixel value
+    event = event && pixelValue > frame[bottomNeighbourRowIndex + leftNeighbourColumn];
     // left
     event = event && pixelValue > frame[rowIndex + leftNeighbourColumn];
     // top left
-    event =
-      event && pixelValue > frame[topNeighbourRowIndex + leftNeighbourColumn];
+    event = event && pixelValue > frame[topNeighbourRowIndex + leftNeighbourColumn];
```

Comment:
The reason these are split over two lines is because we impose an 80 character limit per line. These are probably going to need to be changed back to how they were...

You can see our style rules in the [.clang-format file](https://github.com/OpenChemistry/stempy/blob/master/.clang-format). And you can use the `clang-format` program to automatically format the files for the style rules.

Reply:
This is a [nice script](https://github.com/llvm-mirror/clang/blob/master/tools/clang-format/git-clang-format) that will integrate with git, so you can run `git clang-format`

Reply:
Yeah! `git-clang-format` is great.

Reply:
Thanks! I've installed clang-format, I am sorry that I did not know this.

Reply:
No problem. `git-clang-format` is particularly good because it will only run clang-format on your changes (rather than running clang-format on the whole files).